app/controllers/application_controller.rb

```diff
@@ -31,6 +32,28 @@ def set_current_user
   end
   # rubocop:enable AbcSize
 
+  def login_as_passenger
+    return if @current_user.present?
+
+    @registrant =
+      if session[:passenger_id]
+        Passenger.find_by(id: session[:passenger_id])
+      elsif request.env['fcIdNumber']
+        find_or_initialize_passenger
+      end
+    session[:passenger_id] = @registrant&.id
+  end
+
+  def find_or_initialize_passenger
+    passenger = Passenger.find_or_initialize_by(
+      spire: "#{request.env['fcIdNumber']}@umass.edu",
```

Comment:
Is spire really the `#######@umass.edu?` I thought it was simply the 8 digit number. Is this is a logistics thing?

Reply:
In this database it's saved with the `@umass.edu`. I don't remember why it was done that way.